Added pilot buttons that switch to a panel that displays pilot profil…

maxwellpark opened the pull request:
…es. Uses a prefab & scriptable objects for pilot data. Buttons are created at runtime (similar to mission buttons).

maxwellpark committed:
Added pilot buttons that switch to a panel that displays pilot profiles. Uses a prefab & scriptable objects for pilot data

ambid17 commented on Assets/Scripts/PilotButton.cs:
```diff
@@ -0,0 +1,20 @@
+
+public class PilotButton : MonoBehaviour
+{
+    //public GameObject buttonObject;
```
why is all of this commented out

maxwellpark replied:
It was redundant. Removed it.

ambid17 commented on Assets/Scripts/PilotsManager.cs:
```diff
@@ -0,0 +1,56 @@
+	private void Awake()
+	{
+		pilotProfilePanel.SetActive(false); 
+		pilotProfilePanel.GetComponentInChildren<Button>().onClick.AddListener(delegate { ClosePilotProfilePanel(); });
```
just hold a reference to this button, in case the UI changes

ambid17 replied:
remember to RemoveAllListeners() first

ambid17 commented on Assets/Prefabs/PilotButton.prefab:
```diff
@@ -0,0 +1,201 @@
+%YAML 1.1
```
you need to set the anchors so they scale

ambid17 reviewed (changes requested):
Once we get designs from Fardeen, you will need to make the `PilotsProfilePanel` responsive rather than laying the elements just out on the page. However for a first pass it works well for now :D

maxwellpark committed:
Reworking UI (starting to make responsive and be generated at runtime), and upgraded project version.

maxwellpark committed:
UI generation almost there. Few positioning issues that will be fixed soon.

maxwellpark committed:
Entire pilots UI is now generated at runtime and should be responsive.

The dimension values are stored in the PilotsConstants class so we can easily alter them.

maxwellpark committed:
Deleted redundant class 'PilotButton'

maxwellpark committed:
Merge branch 'develop' into feature/pilots-ui

maxwellpark commented:
Considering making a game-wide UI generation utils class, hence why the PilotsUtils class is unfinished.

ambid17 committed:
Merge branch 'develop' into feature/pilots-ui
